### PR TITLE
feat: update google for api and bug fixes

### PR DESCRIPTION
**Pull request by Claywd**

### CODE REVIEW

File: google-github/cluster-types/mgmt/components/argo-workflows/application.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
...-secure\n  - --auth-mode=client\n  - --auth-mode=sso\n  ingress:\n    enabled: true\n    ingressClassName: nginx\n    annotations:\n      cert-manager.io/cluster-issuer: \"letsencrypt-prod\"\n    hosts:\n      - argo.<DOMAIN_NAME>\n    paths: \n      - /\n    pathType: Prefix\n    tls:\n      - secretName: argo-tls\n        hosts:\n          - argo.<DOMAIN_NAME>\n  sso:\n    issuer: https://vault.<DOMAIN_NAME>/v1/identity/oidc/provider/kubefirst\n    clientId:\n      name: argo-secrets\n      key: client-id\n    clientSecret:\n      name: argo-secrets\n      key: client-secret\n    redirectUrl: https://argo.<DOMAIN_NAME>/oauth2/callback\n    scopes:\n      - email\n      - openid\n      - groups\n      - user\n      - profile\n    # RBAC Config. >= v2.12\n    rbac:\n      enabled: true\nuseDefaultArtifactRepo: true\nuseStaticCredentials: true\nartifactRepository:\n  archiveLogs: false\n  gcs:\n    bucket: <KUBEFIRST_ARTIFACTS_BUCKET>\n    keyFormat: \"argo-workflows/artifacts\\\n      /{{workflow.creationTimestamp.Y}}\\\n      /{{workflow.creationTimestamp.m}}\\\n      /{{workflow.creationTimestamp.d}}\\\n      /{{workflow.uid}}\\\n      /{{workflow.name}}\\\n      /{{pod.name}}\""
```
Comment (johndietz) [inline google-github/cluster-types/mgmt/components/argo-workflows/application.yaml:14]: @Claywd can you confirm this is a good format from your earlier test? if it works i'm good we'll prob want to pretty it up in like 2.3.1+ with some weird pipe yaml +- stuff

File: google-github/cluster-types/mgmt/components/vault/application.yaml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
...om\n  affinity: \"\"\n  ha:\n    enabled: true\n    replicas: 3\n    config: |\n      ui = true\n      listener \"tcp\" {\n        tls_disable     = 1\n        address         = \"[::]:8200\"\n        cluster_address = \"[::]:8201\"\n      }\n      storage \"gcs\" {\n        bucket     = \"<VAULT_DATA_BUCKET>\"\n        ha_enabled = \"true\"\n      }\n      seal \"gcpckms\" {\n        crypto_key  = \"vault-unseal\"\n        key_ring    = \"vault-<CLUSTER_NAME>-<GOOGLE_UNIQUENESS>\"\n        project     = \"<GOOGLE_PROJECT>\"\n        region      = \"global\"\n      }\n      service_registration \"kubernetes\" {}\n  ingress:\n    enabled: true\n    annotations: \n      cert-manager.io/cluster-issuer: \"letsencrypt-prod\"\n    ingressClassName: \"nginx\"\n    pathType: Prefix\n    # When HA mode is enabled and K8s service registration is being used,\n    # configure the ingress to point to the Vault active service.\n    # activeService: true\n    hosts:\n      - host: vault.<DOMAIN_NAME>\n    tls:\n      - secretName: vault-tls\n        hosts:\n          - vault.<DOMAIN_NAME>\n  ui:\n    enabled: true\n    serviceType: \"ClusterIP\"\n    serviceNodePort: null\n    externalPort: 8200"
```
Comment (johndietz) [inline google-github/cluster-types/mgmt/components/vault/application.yaml:19]: same comment

File: google-github/terraform/vault/secrets.tf [hunk 3]
```diff
@@ -123,7 +123,7 @@ resource "vault_generic_secret" "atlantis_secrets" {
       ATLANTIS_GH_WEBHOOK_SECRET          = var.atlantis_repo_webhook_secret,
       GITHUB_OWNER                        = "<GITHUB_OWNER>",
       GITHUB_TOKEN                        = var.github_token,
-      GCP_PROJECT                         = "<GCP_PROJECT>"
+      GOOGLE_PROJECT                         = "<GOOGLE_PROJECT>"
```
Comment (johndietz) [inline google-github/terraform/vault/secrets.tf:126]:
```suggestion
      GOOGLE_PROJECT                      = "<GOOGLE_PROJECT>"
```

File: google-gitlab/cluster-types/mgmt/components/vault/application.yaml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
...om\n  affinity: \"\"\n  ha:\n    enabled: true\n    replicas: 3\n    config: |\n      ui = true\n      listener \"tcp\" {\n        tls_disable     = 1\n        address         = \"[::]:8200\"\n        cluster_address = \"[::]:8201\"\n      }\n      storage \"gcs\" {\n        bucket     = \"<VAULT_DATA_BUCKET>\"\n        ha_enabled = \"true\"\n      }\n      seal \"gcpckms\" {\n        crypto_key  = \"vault-unseal\"\n        key_ring    = \"vault-<CLUSTER_NAME>-<GOOGLE_UNIQUENESS>\"\n        project     = \"<GOOGLE_PROJECT>\"\n        region      = \"global\"\n      }\n      service_registration \"kubernetes\" {}\n  ingress:\n    enabled: true\n    annotations: \n      cert-manager.io/cluster-issuer: \"letsencrypt-prod\"\n    ingressClassName: \"nginx\"\n    pathType: Prefix\n    # When HA mode is enabled and K8s service registration is being used,\n    # configure the ingress to point to the Vault active service.\n    # activeService: true\n    hosts:\n      - host: vault.<DOMAIN_NAME>\n    tls:\n      - secretName: vault-tls\n        hosts:\n          - vault.<DOMAIN_NAME>\n  ui:\n    enabled: true\n    serviceType: \"ClusterIP\"\n    serviceNodePort: null\n    externalPort: 8200"
```
Comment (johndietz) [inline google-gitlab/cluster-types/mgmt/components/vault/application.yaml:19]: same comment here, forgive the repeat, just for thoroughness

File: google-gitlab/cluster-types/mgmt/components/argo-workflows/application.yaml [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
...-secure\n  - --auth-mode=client\n  - --auth-mode=sso\n  ingress:\n    enabled: true\n    ingressClassName: nginx\n    annotations:\n      cert-manager.io/cluster-issuer: \"letsencrypt-prod\"\n    hosts:\n      - argo.<DOMAIN_NAME>\n    paths: \n      - /\n    pathType: Prefix\n    tls:\n      - secretName: argo-tls\n        hosts:\n          - argo.<DOMAIN_NAME>\n  sso:\n    issuer: https://vault.<DOMAIN_NAME>/v1/identity/oidc/provider/kubefirst\n    clientId:\n      name: argo-secrets\n      key: client-id\n    clientSecret:\n      name: argo-secrets\n      key: client-secret\n    redirectUrl: https://argo.<DOMAIN_NAME>/oauth2/callback\n    scopes:\n      - email\n      - openid\n      - groups\n      - user\n      - profile\n    # RBAC Config. >= v2.12\n    rbac:\n      enabled: true\nuseDefaultArtifactRepo: true\nuseStaticCredentials: true\nartifactRepository:\n  archiveLogs: false\n  gcs:\n    bucket: <KUBEFIRST_ARTIFACTS_BUCKET>\n    keyFormat: \"argo-workflows/artifacts\\\n      /{{workflow.creationTimestamp.Y}}\\\n      /{{workflow.creationTimestamp.m}}\\\n      /{{workflow.creationTimestamp.d}}\\\n      /{{workflow.uid}}\\\n      /{{workflow.name}}\\\n      /{{pod.name}}\""
```
Comment (johndietz) [inline google-gitlab/cluster-types/mgmt/components/argo-workflows/application.yaml:14]: same here

**Review by johndietz — APPROVED** [review]

File: google-github/cluster-types/mgmt/components/argo-workflows/application.yaml (hunk 1 above)
Comment (Claywd) [reply google-github/cluster-types/mgmt/components/argo-workflows/application.yaml:14]: yup it works. vscode reformatted I'm guessing.

File: google-gitlab/cluster-types/mgmt/components/vault/application.yaml (hunk 4 above)
Comment (Claywd) [reply google-gitlab/cluster-types/mgmt/components/vault/application.yaml:19]: yeah it works but it looks like vscode squashed it. I can try and unfurl it.

**Pull request merged**